compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partitionning.rs

```diff
@@ -333,6 +368,269 @@ pub mod tests {
+        let mut dag = unparametrized::Dag::new();
+        let input = dag.add_input(16, Shape::number());
+        let variance = get_tfhers_max_noise();
+        let tfhers_partition_in = ExternalPartition {
+            name: String::from("tfhers_in"),
+            macro_params: TFHERS_MACRO_PARAMS,
+            max_variance: variance * 4.0,
+            variance,
+        };
+        let tfhers_partition_out = ExternalPartition {
+            name: String::from("tfhers_out"),
+            macro_params: TFHERS_MACRO_PARAMS,
+            max_variance: variance * 4.0,
+            variance,
+        };
+        let change_part1 = dag.add_change_partition(input, Some(tfhers_partition_in.clone()), None);
+        let lut1 = dag.add_lut(change_part1, FunctionTable::UNKWOWN, 4);
+        let lut2 = dag.add_lut(lut1, FunctionTable::UNKWOWN, 16);
+        let change_part2 = dag.add_change_partition(lut2, None, Some(tfhers_partition_out.clone()));
+
+        let p_cut = PartitionCut::for_each_precision(&dag);
+        let partitions = partitionning_with_preferred(&dag, &p_cut, LOW_PRECISION_PARTITION);
+        assert!(partitions.nb_partitions == 4);
+        let tfhers_partition_index_in =
+            get_external_partition_index_or(&tfhers_partition_in, &p_cut, 2);
+        let tfhers_partition_index_out =
+            get_external_partition_index_or(&tfhers_partition_out, &p_cut, 3);
+        let instrs_partition = partitions.instrs_partition;
+        show_partitionning(&dag, &instrs_partition);
+
+        let consider = |op_i: OperatorIndex| &instrs_partition[op_i.0];
+        assert!(consider(input).instruction_partition == tfhers_partition_index_in);
+        assert!(consider(change_part1).instruction_partition == tfhers_partition_index_in);
+        assert!(consider(change_part1).inputs_transition == [None]);
+        assert!(consider(lut1).instruction_partition == HIGH_PRECISION_PARTITION);
+        assert!(
+            consider(lut1).inputs_transition
+                == [Some(Transition::Internal {
+                    src_partition: tfhers_partition_index_in
+                })]
+        );
+        assert!(consider(lut2).alternative_output_representation.len() == 1);
+        assert!(consider(lut2)
+            .alternative_output_representation
+            .contains(&tfhers_partition_index_out));
+        assert!(consider(lut2).instruction_partition == LOW_PRECISION_PARTITION);
+        assert!(
+            consider(lut2).inputs_transition
+                == [Some(Transition::Internal {
+                    src_partition: HIGH_PRECISION_PARTITION
+                })]
+        );
+        assert!(consider(change_part2).instruction_partition == tfhers_partition_index_out);
+        assert!(
+            consider(change_part2).inputs_transition
+                == [Some(Transition::Additional {
+                    src_partition: LOW_PRECISION_PARTITION
+                })]
+        );
+    }
+
```

Comment:
Maybe adding more cases just to verify no patitionning bug occurs.
Like what happens when add something from external partition with something not.
Eg external_input + lut(other_input) to verify the conversion is added on the lut output.